modules/vaos/app/services/ccra/referral_service.rb

```diff
@@ -65,8 +65,28 @@ def get_referral(id, icn)
         # Note JSON.parse is only used for betamock responses
         data = response.body.is_a?(String) ? JSON.parse(response.body, symbolize_names: true) : response.body
 
-        ReferralDetail.new(data)
+        referral = ReferralDetail.new(data)
+        cache_referral_data(id, referral)
+        referral
       end
     end
+
+    private
+
+    # Caches referral data for use in appointment creation
+    #
+    # @param referral_id [String] The referral ID (consult ID)
+    # @param referral [ReferralDetail] The referral data object
+    # @return [Boolean] True if the cache operation was successful, false if required data is missing or caching fails
+    def cache_referral_data(referral_id, referral)
+      # Delegate to the RedisClient's save_referral_data method
+      eps_redis_client.save_referral_data(referral_id:, referral:)
```

Comment:
Why not just use this one line of code above with a comment? I can see adding a function for multiple lines of code, but just for one it just adds many lines to the file.

Reply:
that's a fair point, there was originally more logic in the method that I moved to the client